source/grid/grid_tools_topology.cc

```diff
@@ -1855,7 +1855,39 @@ namespace GridTools
     const Triangulation<dim, spacedim> &triangulation,
     DynamicSparsityPattern             &cell_connectivity)
   {
-    // Assume at most 16 neighbors per cell (a structured hex mesh has 8)
+    // The choice of 16 or fewer neighbors here is based on empirical
+    // measurements.
+    //
+    // In a structured hex mesh, about 98% of vertices have 16 neighbors or
+    // fewer. Similarly, in an unstructured tetrahedral mesh, if we count the
```

Comment:
Do you mean
```suggestion
    // In a structured tet mesh, about 98% of vertices have 16 neighbors or
    // fewer. Similarly, in an unstructured tetrahedral mesh, if we count the
```
(I have not had a hex mesh, even unstructured one, with more than 16 neighbors.)

Reply:
Fixed!